lib/wallet/src/Cardano/Wallet/Api/Server/Handlers/TxCBOR.hs

```diff
@@ -0,0 +1,89 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE TypeOperators #-}
+
+-- |
+-- Copyright: © 2020 IOHK
+-- License: Apache-2.0
+--
+
+module Cardano.Wallet.Api.Server.Handlers.TxCBOR
+  ( parseTxCBOR
+  , ParsedTxCBOR (..)
+  )
+  where
+
+import Prelude hiding
+    ( (.) )
+
+import Cardano.Wallet.Api.Server.Error
+    ( IsServerError (..), apiError, liftE, showT )
+import Cardano.Wallet.Api.Types
+    ( ApiErrorCode (UnexpectedError) )
+import Cardano.Wallet.Primitive.Types
+    ( Certificate )
+import Cardano.Wallet.Read
+    ( Tx (..) )
+import Cardano.Wallet.Read.Eras
+    ( (:.:)
+    , EraFun (..)
+    , K (..)
+    , applyEraFun
+    , extractEraValue
+    , sequenceEraValue
+    , (*.**)
+    )
+import Cardano.Wallet.Read.Eras.EraFun
+    ( EraFunK (..) )
+import Cardano.Wallet.Read.Tx.CBOR
+    ( TxCBOR, deserializeTx )
+import Cardano.Wallet.Read.Tx.Certificates
+    ( getEraCertificates )
+import Codec.CBOR.Read
+    ( DeserialiseFailure )
+import Control.Category
+    ( (.) )
+import GHC.Generics
+    ( Generic )
+import Servant.Server
+    ( Handler, err500 )
+
+import qualified Cardano.Wallet.Read.Primitive.Tx.Features.Certificates as Feature
+import qualified Data.ByteString.Lazy as BL
+
+newtype ErrParseCBOR = ErrParseCBOR DeserialiseFailure
+    deriving (Eq, Show)
+
+instance IsServerError ErrParseCBOR where
+    toServerError (ErrParseCBOR df) =
+        apiError err500 UnexpectedError $ mconcat
+            [ "Error while trying to parse a transaction CBOR from the database"
+            , showT df
+            ]
+
+-- | Values parsed out of a CBOR for a 'Tx' in any era
+newtype ParsedTxCBOR = ParsedTxCBOR
+    { certificates :: [Certificate]
+    }
```

Comment:
This type will eventually become equivalent to the old `Primitive.Tx` type in that it tries to merge all the transaction features for the purpose of presenting them in the HTTP API.

I think that the name `ParsedTxCBOR` is not good, because it does not reflect the purpose. We may want to change that in the future. Perhaps `ApiTx` would be appropriate, even.

In any case, this type is still better than before, in that the type is close to the HTTP API and removed from the database, but the issue of merging information over different eras remains. We may want to reconsider this in the future.